### PR TITLE
Migrate to junit5

### PR DESCRIPTION
**Pull request by rtroilo**
Migrating from junit4 to junit5. 

The migrations was mostly straight forward, changing import from 
`org.junit` to `org.junit.jupiter.api`.

In the `oshdb-util` module there is a dependency to `json-simple` which includes `junit4` as a compile dependency. But if you check the github project the scope of the junit4 dependency is reduced to `test` as it should be. So in this PR we exculde the `junit4` dependency of `json-simple`.

There are some test which checks explicit that an exception is NOT thrown. Like
```
try {
  OSHDBGeometryBuilder.getGeometry(entity1, timestamp, areaDecider);
} catch (Exception e) {
  e.printStackTrace();
  fail("Should not have thrown any exception");
}
```
which is now replaces with:
```
assertDoesNotThrow(() -> {
  OSHDBGeometryBuilder.getGeometry(entity1, timestamp, areaDecider);
});
```
We could discuss if this check is actually necessary.

### Checklist
- [x] My code follows the [code-style](https://github.com/GIScience/oshdb/blob/master/CONTRIBUTING.md) rules, and I have checked on the [static analyses](https://jenkins.ohsome.org/job/oshdb/view/change-requests/) and [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) (if applicable) results
- [x] I have commented my code
- ~[ ] I have written javadoc (required for public classes and methods)~
- ~[ ] I have added sufficient unit tests~
- [ ] I have made corresponding changes to the [documentation](https://github.com/GIScience/oshdb/tree/master/documentation)
- [ ] I have updated the [CHANGELOG.md](https://github.com/GIScience/oshdb/blob/master/CHANGELOG.md)
- ~[ ] I have adjusted the [examples](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-examples) or [created an issue](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-examples/-/issues/new) in the corresponding repository~
- ~[ ] I have adjusted the [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) or [created an issue](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-benchmarks/-/issues/new) in the corresponding repository~

Please check all finished tasks. If some tasks do not apply to your PR, please cross their text out (by using `~...~`) and remove their checkboxes.

### CODE REVIEW

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 1]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
-      fail("Should not have thrown any exception");
-    }
+    });
+
     // third version
     OSMEntity entity2 = testData.relations().get(500L).get(2);
     OSHDBTimestamp timestamp2 = new OSHDBTimestamp(entity2);
-    try {
+
+
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:59]: Additional empty line on purpose?

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 2]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
-    Geometry result;
-    try {
+
+    assertDoesNotThrow(() -> {
       OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-      result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
+      Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
       assertTrue(result.isEmpty());
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:110]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 3]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
     // relation tags changing
     OSMEntity entity = testData.relations().get(503L).get(0);
     OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
       assertEquals(1, result.getNumGeometries());
       assertTrue(result.getGeometryN(0) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:124]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 4]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
     // second version
     OSMEntity entity1 = testData.relations().get(503L).get(1);
     OSHDBTimestamp timestamp1 = new OSHDBTimestamp(entity1);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result1 = OSHDBGeometryBuilder.getGeometry(entity1, timestamp1, areaDecider);
       assertTrue(result1 instanceof GeometryCollection);
       assertTrue(result1.isValid());
       assertEquals(1, result1.getNumGeometries());
       assertTrue(result1.getGeometryN(0) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:134]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 5]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
     // third version
     OSMEntity entity2 = testData.relations().get(503L).get(2);
     OSHDBTimestamp timestamp2 = new OSHDBTimestamp(entity2);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result2 = OSHDBGeometryBuilder.getGeometry(entity2, timestamp2, areaDecider);
       assertTrue(result2 instanceof GeometryCollection);
       assertTrue(result2.isValid());
       assertEquals(1, result2.getNumGeometries());
       assertTrue(result2.getGeometryN(0) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:144]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 6]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
     OSMEntity entity = testData.relations().get(504L).get(0);
     OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
       assertEquals(2, result.getNumGeometries());
       assertTrue(result.getGeometryN(0) instanceof LineString);
       assertTrue(result.getGeometryN(1) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:159]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 7]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
     OSMEntity entityAfter = testData.relations().get(505L).get(0);
     OSHDBTimestamp timestampAfter =  TimestampParser.toOSHDBTimestamp("2012-02-01T00:00:00Z");
-    try {
+    assertDoesNotThrow(() -> {
       Geometry resultAfter = OSHDBGeometryBuilder
           .getGeometry(entityAfter, timestampAfter, areaDecider);
       assertTrue(resultAfter instanceof GeometryCollection);
       assertTrue(resultAfter.isValid());
       assertEquals(1, resultAfter.getNumGeometries());
       assertTrue(resultAfter.getGeometryN(0) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:207]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 8]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
     // relation getting more ways, one disappears, last version not valid
     OSMEntity entity = testData.relations().get(500L).get(0);
     OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:43]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 9]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result1 = OSHDBGeometryBuilder.getGeometry(entity1, timestamp1, areaDecider);
       assertTrue(result1 instanceof GeometryCollection);
       assertTrue(result1.isValid());
-    } catch (Exception e) {
-      e.printStackTrace();
-      fail("Should not have thrown any exception");
-    }
+    });
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:53]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 10]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
     // third version
     OSMEntity entity2 = testData.relations().get(500L).get(2);
     OSHDBTimestamp timestamp2 = new OSHDBTimestamp(entity2);
-    try {
+
+
+    assertDoesNotThrow(() -> {
       Geometry result2 = OSHDBGeometryBuilder.getGeometry(entity1, timestamp2, areaDecider);
       assertTrue(result2 instanceof GeometryCollection || result2 instanceof Polygonal);
       assertEquals(3, result2.getNumGeometries());
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:63]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 11]
```diff
@@ -37,239 +37,195 @@ public void testGeometryChange() {
...
-      fail("Should not have thrown any exception");
-    }
+    });
     // second version
     OSMEntity entity1 = testData.relations().get(501L).get(1);
     OSHDBTimestamp timestamp1 = new OSHDBTimestamp(entity1);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result1 = OSHDBGeometryBuilder.getGeometry(entity1, timestamp1, areaDecider);
       assertTrue(result1.isEmpty());
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:85]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 12]
```diff
@@ -278,42 +234,33 @@ public void testGeometryChangeOfNodeCoordinatesInRelationAndWay() {
...
     OSMEntity entity = testData.relations().get(508L).get(0);
-    Geometry result;
-    try {
+
+    assertDoesNotThrow(() -> {
       OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-      result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
+      Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:262]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 13]
```diff
@@ -322,277 +269,223 @@ public void testVisibleChangeOfNodeInWay() {
...
     // relation, way member: node 53 changes tags, 51 changes coordinates
     OSMEntity entity = testData.relations().get(510L).get(0);
     OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
       assertEquals(1, result.getNumGeometries());
       assertTrue(result.getGeometryN(0) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:303]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 14]
```diff
@@ -322,277 +269,223 @@ public void testVisibleChangeOfNodeInWay() {
...
+    });
     // version after, visible false
     OSMEntity entityAfter = testData.relations().get(511L).get(0);
     OSHDBTimestamp timestampAfter =  TimestampParser.toOSHDBTimestamp("2017-02-01T00:00:00Z");
-    try {
+    assertDoesNotThrow(() -> {
       Geometry resultAfter = OSHDBGeometryBuilder.getGeometry(entityAfter, timestampAfter,
           areaDecider);
       assertTrue(resultAfter instanceof GeometryCollection);
       assertTrue(resultAfter.isEmpty());
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:337]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 15]
```diff
@@ -322,277 +269,223 @@ public void testVisibleChangeOfNodeInWay() {
...
     OSMEntity entity = testData.relations().get(512L).get(0);
     OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
       assertEquals(2, result.getNumGeometries());
       assertTrue(result.getGeometryN(0) instanceof LineString);
       assertTrue(result.getGeometryN(1) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:352]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 16]
```diff
@@ -322,277 +269,223 @@ public void testVisibleChangeOfNodeInWay() {
...
     // relation, way member: way 122 changes tags
     OSMEntity entity = testData.relations().get(513L).get(0);
     OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertTrue(result.isValid());
       assertEquals(1, result.getNumGeometries());
       assertTrue(result.getGeometryN(0) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:377]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 17]
```diff
@@ -322,277 +269,223 @@ public void testVisibleChangeOfNodeInWay() {
...
     OSMEntity entity1 = testData.relations().get(515L).get(1);
     OSHDBTimestamp timestamp1 = new OSHDBTimestamp(entity1);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result1 = OSHDBGeometryBuilder.getGeometry(entity1, timestamp1, areaDecider);
       assertTrue(result1 instanceof GeometryCollection);
       assertTrue(result1.isValid());
       assertEquals(2, result1.getNumGeometries());
       assertTrue(result1.getGeometryN(0) instanceof LineString);
       assertTrue(result1.getGeometryN(1) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:447]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java [hunk 18]
```diff
@@ -322,277 +269,223 @@ public void testVisibleChangeOfNodeInWay() {
...
     OSMEntity entity2 = testData.relations().get(519L).get(1);
     OSHDBTimestamp timestamp2 = new OSHDBTimestamp(entity2);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result2 = OSHDBGeometryBuilder.getGeometry(entity2, timestamp2, areaDecider);
       assertTrue(result2 instanceof GeometryCollection);
       assertEquals(3, result2.getNumGeometries());
       assertTrue(result2.getGeometryN(0) instanceof LineString);
       assertTrue(result2.getGeometryN(1) instanceof LineString);
       assertTrue(result2.getGeometryN(2) instanceof LineString);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationNotMultipolygonTest.java:487]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationTest.java [hunk 19]
```diff
@@ -67,14 +67,11 @@ public void testGeometryChange() throws ParseException {
     // third version
     OSMEntity entity2 = testData.relations().get(500L).get(2);
     OSHDBTimestamp timestamp2 = new OSHDBTimestamp(entity2);
-    try {
+    assertDoesNotThrow(() -> {
       Geometry result2 = OSHDBGeometryBuilder.getGeometry(entity1, timestamp2, areaDecider);
       assertTrue(result2 instanceof GeometryCollection || result2 instanceof Polygonal);
       assertEquals(3, result2.getNumGeometries());
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationTest.java:73]: Can be moved outside of the assert block.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationTest.java [hunk 20]
```diff
@@ -272,29 +266,23 @@ public void testGeometryCollection() {
     // relation, not valid, should be a not empty geometryCollection
     // https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb/issues/143
     OSMEntity entity = testData.relations().get(507L).get(0);
-    try {
+    assertDoesNotThrow(() -> {
       OSHDBTimestamp timestamp = new OSHDBTimestamp(entity);
       Geometry result = OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
       assertTrue(result instanceof GeometryCollection);
       assertEquals(6, result.getNumGeometries());
       assertFalse(result instanceof MultiPolygon);
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataRelationTest.java:274]: Can be moved outside of the assert block.

**Review by joker234 — COMMENTED** [review]
Thanks! I made a few minor remarks.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmtestdata/OSHDBGeometryBuilderTestOsmTestData3xxTest.java [hunk 21]
```diff
@@ -19,74 +18,81 @@
...
     // user name is not priority
     int entityUid = testData.nodes().get(202000L).get(0).getUserId();
     assertTrue(entityUid < 1);
   }
 
   @Test
-  public void test303() {
+  void test303() {
     // uid 0 is the anonymous user
     int entityUid = testData.nodes().get(203000L).get(0).getUserId();
```
Comment (joker234) [inline oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmtestdata/OSHDBGeometryBuilderTestOsmTestData3xxTest.java:62]:
```suggestion
    int entityUid = nodes(203000L, 0).getUserId();
```

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmtestdata/OSHDBGeometryBuilderTestOsmTestData1xxTest.java [hunk 22]
```diff
@@ -87,38 +77,33 @@ public void test111() {
...
-  public void test114() {
+  void test114() {
     // Two ways connected end-to-beginning
-    OSMEntity entity1 = testData.ways().get(114800L).get(0);
-    OSMEntity entity2 = testData.ways().get(114801L).get(0);
     TagInterpreter areaDecider = new FakeTagInterpreterAreaMultipolygonAllOuters();
-    Geometry result1 = OSHDBGeometryBuilder.getGeometry(entity1, timestamp, areaDecider);
-    Geometry result2 = OSHDBGeometryBuilder.getGeometry(entity2, timestamp, areaDecider);
+    Geometry result1 = OSHDBGeometryBuilder.getGeometry(ways(114800L, 0), timestamp, areaDecider);
+    Geometry result2 = OSHDBGeometryBuilder.getGeometry(ways(114801L, 0), timestamp, areaDecider);
```
[the remark this answers is not in the document]
Comment (rtroilo) [reply oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmtestdata/OSHDBGeometryBuilderTestOsmTestData1xxTest.java:106]: Beware that it this case a different areaDecider is used!

File: oshdb-api/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBJdbcMissingTables.java [hunk 23]
```diff
@@ -1,57 +1,71 @@
...
   }
 
   @Override
-  @Test(expected = OSHDBTableNotFoundException.class)
-  public void testOSMContributionViewStream() throws Exception {
-    super.testOSMEntitySnapshotView();
+  @Test()
+  void testOSMContributionViewStream() {
+    assertThrows(OSHDBTableNotFoundException.class, () -> {
+      super.testOSMEntitySnapshotView();
```
Comment (joker234) [inline oshdb-api/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBJdbcMissingTables.java:44]:
```suggestion
      super.testOSMContributionViewStream();
```

File: oshdb-api/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBJdbcMissingTables.java [hunk 24]
```diff
@@ -1,57 +1,71 @@
...
   }
 
   @Override
-  @Test(expected = OSHDBTableNotFoundException.class)
-  public void testOSMEntitySnapshotViewStream() throws Exception {
-    super.testOSMEntitySnapshotView();
+  @Test()
+  void testOSMEntitySnapshotViewStream() {
+    assertThrows(OSHDBTableNotFoundException.class, () -> {
+      super.testOSMEntitySnapshotView();
```
Comment (joker234) [inline oshdb-api/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBJdbcMissingTables.java:52]:
```suggestion
      super.testOSMEntitySnapshotViewStream();
```

**Review by joker234 — COMMENTED** [review]
Two additional things I found

File: oshdb-api-ignite/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBIgniteMissingCache.java [hunk 25]
```diff
@@ -1,56 +1,70 @@
...
   }
 
   @Override
-  @Test(expected = OSHDBTableNotFoundException.class)
-  public void testOSMContributionViewStream() throws Exception {
-    super.testOSMEntitySnapshotView();
+  @Test()
+  void testOSMContributionViewStream() throws Exception {
+    assertThrows(OSHDBTableNotFoundException.class, () -> {
+      super.testOSMEntitySnapshotView();
```
Comment (joker234) [inline oshdb-api-ignite/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBIgniteMissingCache.java:43]:
```suggestion
      super.testOSMContributionViewStream();
```

File: oshdb-api-ignite/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBIgniteMissingCache.java [hunk 26]
```diff
@@ -1,56 +1,70 @@
...
   }
 
   @Override
-  @Test(expected = OSHDBTableNotFoundException.class)
-  public void testOSMEntitySnapshotViewStream() throws Exception {
-    super.testOSMEntitySnapshotView();
+  @Test()
+  void testOSMEntitySnapshotViewStream() throws Exception {
+    assertThrows(OSHDBTableNotFoundException.class, () -> {
+      super.testOSMEntitySnapshotView();
```
Comment (joker234) [inline oshdb-api-ignite/src/test/java/org/heigit/ohsome/oshdb/api/tests/TestMapReduceOSHDBIgniteMissingCache.java:51]:
```suggestion
      super.testOSMEntitySnapshotViewStream();
```

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataNodesTest.java [hunk 27]
```diff
@@ -1,99 +1,89 @@
...
-    OSHDBTimestamp timestampAfter =  TimestampParser.toOSHDBTimestamp("2012-01-01T00:00:00Z");
-    Geometry resultAfter = OSHDBGeometryBuilder.getGeometry(entity3, timestampAfter, areaDecider);
-    assertTrue(resultAfter instanceof Point);
-    assertEquals(1.425, ((Point) resultAfter).getX(), DELTA);
-    assertEquals(1.23, ((Point) resultAfter).getY(), DELTA);
+    result = buildGeometry(nodes(1L, 2), "2012-01-01T00:00:00Z");
+    assertTrue(result instanceof Point);
+    assertEquals(1.425, ((Point) result).getX(), DELTA);
+    assertEquals(1.23, ((Point) result).getY(), DELTA);
   }
```
[the remark this answers is not in the document]
Comment (rtroilo) [reply oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataNodesTest.java:50]: I agree with you

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataNodesTest.java [hunk 28]
```diff
@@ -1,99 +1,89 @@
...
-    assertTrue(result2 instanceof Point);
-    assertEquals(1.43, ((Point) result2).getX(), DELTA);
-    assertEquals(1.24, ((Point) result2).getY(), DELTA);
-    OSMEntity entity3 = testData.nodes().get(2L).get(2);
-    OSHDBTimestamp timestamp3 = new OSHDBTimestamp(entity3.getEpochSecond());
-    Geometry result3 = OSHDBGeometryBuilder.getGeometry(entity3, timestamp3, areaDecider);
-    assertTrue(result3 instanceof Point);
-    assertEquals(1.43, ((Point) result3).getX(), DELTA);
-    assertEquals(1.24, ((Point) result3).getY(), DELTA);
   }
```
[the remark this answers is not in the document]
Comment (rtroilo) [reply oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataNodesTest.java:79]: yes we should start to use them in up coming test more often.

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataWaysTest.java [hunk 29]
```diff
@@ -40,194 +33,154 @@ private static void checkLineString(double[][] expectedCoordinates, Geometry res
...
-    OSHDBTimestamp timestamp3 = new OSHDBTimestamp(entity3);
-    Geometry result3 = OSHDBGeometryBuilder.getGeometry(entity3, timestamp3, areaDecider);
-    assertTrue(result3 instanceof LineString);
-    assertEquals(5, result3.getNumPoints());
+    result = buildGeometry(ways(103L, 1));
+    assertTrue(result instanceof LineString);
+    assertEquals(5, result.getNumPoints());
 
+    // last appearance
+    result = buildGeometry(ways(103L, 2));
```
[the remark this answers is not in the document]
Comment (rtroilo) [reply oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmhistorytestdata/OSHDBGeometryBuilderTestOsmHistoryTestDataWaysTest.java:108]: I think so

File: oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmtestdata/OSHDBGeometryBuilderTestOsmTestData3xxTest.java [hunk 30]
```diff
@@ -19,74 +18,81 @@
...
-    areaDecider = new OSMXmlReaderTagInterpreter(testData);
+  OSHDBGeometryBuilderTestOsmTestData3xxTest() {
+    super("./src/test/resources/osm-testdata/all.osm");
   }
 
   private Geometry buildEntityGeometry(long id) {
-    OSMEntity entity = testData.nodes().get(id).get(0);
+    OSMEntity entity = nodes(id, 0);
     return OSHDBGeometryBuilder.getGeometry(entity, timestamp, areaDecider);
   }
```
[the remark this answers is not in the document]
Comment (joker234) [reply oshdb-util/src/test/java/org/heigit/ohsome/oshdb/util/geometry/osmtestdata/OSHDBGeometryBuilderTestOsmTestData3xxTest.java]: see 4cc312ca

**Review by joker234 — DISMISSED** [review]

**Review by joker234 — APPROVED** [review]

**Pull request merged**